src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,615 @@
+            me->SummonCreature(NPC_FROZEN_CORE, SummonPositions[0], TEMPSUMMON_CORPSE_DESPAWN);
+        }
+
+        void Reset() override
+        {
+            Initialize();
+            me->SetUInt32Value(UNIT_NPC_EMOTESTATE, 0);
+            me->HandleEmoteCommand(EMOTE_ONESHOT_EMERGE);
+        }
+        
+        void EnterCombat(Unit* who) override
+        {
+            DoZoneInCombat();
+            _events.SetPhase(PHASE_ONE);
+            _events.ScheduleEvent(EVENT_COLD_SLAP, 500, 0,PHASE_ONE);
+            _events.ScheduleEvent(EVENT_SUMMON_HAILSTONE, 1000, 0,PHASE_ONE);
+            _events.ScheduleEvent(EVENT_SUMMON_COLDWEAVE, 8000, 0,PHASE_ONE);
+            _events.ScheduleEvent(EVENT_ICE_SPEAR, 9500, 0,PHASE_ONE);
+            _events.ScheduleEvent(EVENT_EMOTE_EARTHEN_ATTACK, 12000, 0, PHASE_ONE);
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            instance->DoCastSpellOnPlayers(SPELL_AHUNE_ACHIEVEMENT);
+		    DoCast(me, SPELL_SUMMON_LOOT);
+            if (Creature* ahuneBunny = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_AHUNE_BUNNY)))
+                if (ahuneBunny)
+                    me->Kill(ahuneBunny);
+            _summons.DespawnAll();
+
+            Map::PlayerList const& players = me->GetMap()->GetPlayers();
+            if (!players.isEmpty())
+            {
+                if (Group* group = players.begin()->GetSource()->GetGroup())
+                    if (group->isLFGGroup())
+                        sLFGMgr->FinishDungeon(group->GetGUID(), 286);
+            }
+        }
+
+        void DamageTaken(Unit* who, uint32& damage) override
+        {
+            if (Creature* core = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_FROZEN_CORE)))
+            {
+                me->CastSpell(core, SPELL_SYNCH_HEALTH);
+                if (core->GetHealth() > damage)
+                    core->SetHealth(core->GetHealth() - damage);
+                else
+                    core->CastSpell(core, SPELL_SUICIDE);
+            }
+        }
+
+        void JustSummoned(Creature* summon) override
+        {
+            switch (summon->GetEntry())
+            {
+            case NPC_FROZEN_CORE:
+                summon->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_DISABLE_MOVE);
+                summon->ApplySpellImmune(0, IMMUNITY_EFFECT, SPELL_EFFECT_KNOCK_BACK, true);
+                summon->ApplySpellImmune(0, IMMUNITY_MECHANIC, MECHANIC_GRIP, true);
+                summon->SetReactState(REACT_PASSIVE);
+                break;
+            case NPC_SLIPPERY_FLOOR:
+                summon->SetReactState(REACT_PASSIVE);
+                summon->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE | UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_DISABLE_MOVE);
+                me->AddAura(SPELL_SLIPPERY_FLOOR_AMBIENT, summon);
+                break;
+            case NPC_ICE_SPEAR_BUNNY:
```

Comment:
This case is redundant since you're calling 

```cpp
summon->SetInCombatWithZone();
```

Few lines under it, outside the switch block